### ISSUE TITLE
feature: Wordpress Hook On /auth endpoint BEFORE authentication happens -> Prevent Brute Force Attacks

### ISSUE THREAD

**Issue by jkdigital-au** [nicumicle/simple-jwt-login#59]
Howdy! I'm using this plugin for WP Rest API Authentication & SSO Login Implementation.

It works wonderfully,  but I want to add some protection to the /auth endpoint for bruteforce attacks. 

On my login form,  I send a POST request to the /auth endpoint to get a valid JWT. This works, but I currently don't have measures to prevent Brute Force Attacks. My login form is on the client side (in a react app) so I need something on the API side to trigger.

I've already put together the code to get this working. I really just want a hook I can use to check the login attempts for a given user, before I continue with trying to authenticate the user.

"_simple_jwt_login_response_auth_user_"

This is the hook I tried, but it runs only when authentication is successful, not on every response, so is not functional for me.

**TLDR**: Need a hook that fires on the /auth endpoint BEFORE authentication commences to track login attempts and prevent Brute Force Attacks.

**Comment by nicumicle** [comment]
Hello @jkdigital-au,

I've just added the new hook. 
The new hook is called before the initialization of each endpoint. You can read more here: https://simplejwtlogin.com/docs/hooks#16-simple_jwt_login_before_endpoint


Also, I've added some examples in the documentation: 
https://simplejwtlogin.com/docs/hooks#block-requests-on-auth-for-a-specific-email-address

**Comment by jkdigital-au** [comment]
Amazing, thank you so much for your hard work and dedication!
